app/routes.py

```diff
@@ -1,2 +1,289 @@
+        return jsonify({"details": "Invalid data"}), 400
+
+    new_task = Task(title=request_body["title"],
+                    description=request_body["description"],
+                    completed_at=request_body["completed_at"])
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    response_body = {
+        "task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": new_task.completed_at is not None
+        }
+    }
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])
+def read_task():
+
+    # tasks hold all the information from the database. Becasue Task is referring to a table. So we are capturing all the records(rows) in the table and holding it in tasks.
+    # queries are the records (i.e. rows)
+    # creating task_response variable
+    title_query = request.args.get("sort")
+    if title_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif title_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+
+    task_responses = []
+    for task in tasks:
+        task_response = {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+        if task.goal_id:
+            task_response["goal_id"] = task.goal_id
+        task_responses.append(task_response)
+    return jsonify(task_responses), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    task_response = {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": task.completed_at is not None
+    }
+    if task.goal_id:
+        task_response["goal_id"] = task.goal_id
+
+    response_body = {"task": task_response}
+
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    response_body = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+    }
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return jsonify({
+        "details": f"Task {task.task_id} \"{task.title}\" successfully deleted"
+    }), 200
+
+
+def slack_bot(title):
```

Comment:
👍 

Nice job putting the Slack calling code into its own helper function. All the mark complete endpoint really needs to know how to do is call a method that can post to Slack with the message to post. The rest of the details are constrained to the helper, and potentially could be called from multiple places!